manifests/charts/ai-gateway-helm/templates/_helpers.tpl

```diff
@@ -68,6 +68,33 @@ Create the name of the service account to use
 {{- end }}
 {{- end }}
 
+{{/*
+Create the name of the cluster role to use
+*/}}
+{{- define "ai-gateway-helm.controller.clusterRoleName" -}}
+{{- if .Values.controller.clusterRole.suffixClusterRole }}
+{{- printf "%s:%s" (default (include "ai-gateway-helm.controller.fullname" .) .Values.controller.serviceAccount.name) .Release.Namespace }}
+{{- else }}
+{{- default (include "ai-gateway-helm.controller.fullname" .) .Values.controller.serviceAccount.name }}
+{{- end }}
+{{- end }}
+
+{{- define "ai-gateway-helm.inference-pool.clusterRoleName" -}}
+{{- if .Values.controller.clusterRole.suffixClusterRole }}
+{{- printf "%s:%s" "envoy-ai-gateway-inference-pool-reader" .Release.Namespace}}
+{{- else }}
+{{- "envoy-ai-gateway-inference-pool-reader" }}
+{{- end}}
+{{- end -}}
+
+{{- define "ai-gateway-helm.inference-pool.clusterRoleBindingName" -}}
+{{- if .Values.controller.clusterRole.suffixClusterRole }}
+{{- printf "%s:%s" "envoy-ai-gateway-inference-pool-reader-binding" .Release.Namespace}}
+{{- else }}
+{{- "envoy-ai-gateway-inference-pool-reader-binding" }}
+{{- end}}
+{{- end -}}
+
```

Comment:
ClusterRoles won't change across installations, so it probably doesn't make sense to have N cluster roles named after the namespace.

Instead, I would create a consistent cluster role name, not related to the service account, and attach the service accounts to it. It will be much cleaner. WDYT?

Reply:
yeah good point, let's do only for role bindings (is that what you suggest)

Reply:
Yes. ClusterRole should be unique, and rolebindings should bind the ServiceAccount in each namespace to that ClusterRole.